dali/operators/decoder/video/video_decoder_base.h

```diff
@@ -18,118 +18,225 @@
-   * will return a partial sequence.
-   *
-   * @param output The SampleView in which the decoded sequence will be put.
-   * @param sample_idx Index of the encoded video that shall be decoded.
-   * @param num_frames How many frames shall be decoded.
-   * @param pad_value What to (optionally) pad the partial sequence with.
-   * @return False, if less than `num_frames` have been decoded.
-   */
-  bool DecodeFrames(SampleView<OutBackend> output, int64_t sample_idx, int64_t num_frames,
-                    std::optional<SampleView<OutBackend>> pad_value = std::nullopt,
-                    std::optional<cudaStream_t> stream = std::nullopt) {
+  bool DecodeFrames(SampleView<OutBackend> output, int64_t sample_idx,
+                    int64_t start_frame, int64_t sequence_length, int64_t stride,
+                    cudaStream_t stream = 0) {
     auto &frames_decoder = *frames_decoders_[sample_idx];
     int64_t frame_size = frames_decoder.FrameSize();
-
-    DALI_ENFORCE(!pad_value.has_value() || pad_value->shape().num_elements() == frame_size,
-                 make_string("Provided pad_value has improper number of elements. Expected: ",
-                             frame_size, "; Actual: ", pad_value->shape().num_elements()));
-
     uint8_t *output_data = output.template mutable_data<uint8_t>();
 
     int64_t f = 0;
-    // Work until:
-    //    (a) There are no more frames, or
-    //    (b) Sufficient number of frames has been decoded.
-    for (; f < num_frames && frames_decoder.NextFrameIdx() != -1; f++) {
-      frames_decoder.ReadNextFrame(output_data + f * frame_size);
-    }
-    assert(f <= num_frames);
-    bool full_sequence_decoded = f == num_frames;
-    // If there's an insufficient number of frames, pad if requested.
-    for (; f < num_frames && pad_value.has_value(); f++) {
-      kernels::copy<storage_backend_for_copy_kernel, storage_backend_for_copy_kernel>(
-              output_data + f * frame_size, pad_value->raw_data(), frame_size,
-              std::is_same_v<storage_backend_for_copy_kernel, StorageGPU> ? *stream : 0);
+    if (start_frame > 0) {
+      frames_decoder.SeekFrame(start_frame);
     }
-    return full_sequence_decoded;
-  }
 
+    // Read frames with specified stride
+    for (; f < sequence_length && frames_decoder.NextFrameIdx() != -1; f++) {
+      frames_decoder.ReadNextFrame(output_data + f * frame_size);
+      for (int i = 1; i < stride && frames_decoder.NextFrameIdx() != -1; i++) {
+        frames_decoder.ReadNextFrame(nullptr, false);
+      }
+    }
 
-  /**
-   * @brief Decode sample with index `idx` to `output` tensor.
-   */
-  void DecodeSample(SampleView<OutBackend> output, int64_t idx) {
-    int64_t num_frames = output.shape()[0];
-    DecodeFrames(output, idx, num_frames);
+    // Handle padding if needed
+    if (f < sequence_length) {
+      switch (padding_mode_) {
+        case PaddingMode::kNone:
+          return false;
```

Comment:
I wonder if that is what we want to do? We return garbage frames. Maybe we can return shorter sequence for this sample (so adjust the length in SetupImpl accordingly)?